### PR TITLE
[bugfix] Fix PPMissingLayer AttributeError when Using PP

### PR DESCRIPTION
**Pull request by RolaoDenthu**
cc @HaiShaw 

## Motivation

When using PP, accessing attributes like `layer.mlp` or `embed_tokens.embedding_dim` on `PPMissingLayer` raises `AttributeError` and crashes model loading.

## Solution

This PR adds `__getattr__` and `__setattr__` to `PPMissingLayer` so that attribute access returns `self` instead of crashing. This way `isinstance(layer.mlp, SomeMoE)` safely returns `False`, and no per-model file changes are needed.

## Accuracy Tests

Tested pp on  2 nodes × 8 MI355X GPUs using DeepSeek-V3-0324, with setting  PP=2/TP=8

| Scenario | Throughput (tok/s) | TTFT | TPOT | Median ITL |
|----------|-------------------|------|------|------------|
| Max throughput (500 reqs, inf rate) | 5,887 | - | 193ms | 148ms |
| Online serving (300 reqs, 8 req/s) | 5,128 | 124ms | 120ms | 112ms |
| Single request | - | 98ms | 29.5ms | 29.5ms |

PP=4/TP=4 crashes during CUDA graph capture because TileLang NSA kernel requests ~218KB shared memory, exceeding the MI355X per-workgroup LDS limit of 160KB.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Review by HaiShaw — APPROVED** [review]

File: python/sglang/srt/layers/utils/common.py [hunk 1]
```diff
@@ -59,12 +59,28 @@ class PPMissingLayer(torch.nn.Identity):
...
 
     def __init__(self, *args, **kwargs):
         super().__init__()
         self.return_tuple = kwargs.get("return_tuple", False)
 
+    def __getattr__(self, name: str):
+        try:
+            return super().__getattr__(name)
+        except (AttributeError, KeyError):
+            return self
```
Comment (ch-wan) [inline python/sglang/srt/layers/utils/common.py:76]: Should we return `None` here if key does not exist? We may have some usage like `getattr(module, "quant_method")`. Example: https://github.com/sgl-project/sglang/blob/6dfa8a40bc6a18512677b4cc60a42bf76fadea68/python/sglang/multimodal_gen/runtime/loader/fsdp_load.py#L145

Comment (RolaoDenthu) [reply python/sglang/srt/layers/utils/common.py:76]: Thank you for the catch! I added return self here because `PPMissingLayer `replaces layers missing on the current GPU rank during PP. Returning `None ` would break chained access like `layer.mlp.qkv_proj`. But you're right that this breaks `getattr(module, "quant_method", None)` and potentially other `getattr`. Fixed in https://github.com/sgl-project/sglang/pull/21114.
